### PR TITLE
Add spmi flash driver

### PR DESCRIPTION
**Pull request by alikates**
pmi8950 seems to be using the same flash subsystem as pm8941 so I added a driver which was submitted for upstreaming but hasn't been merged yet.
I updated our pmi8950.dtsi to use the new driver.

### CODE REVIEW

File: Documentation/devicetree/bindings/leds/leds-qcom-spmi-flash.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
```
Comment (okias) [inline Documentation/devicetree/bindings/leds/leds-qcom-spmi-flash.yaml:1]: `GPL-2.0-only`

File: drivers/leds/flash/leds-qcom-spmi-flash.c [hunk 2]
```diff
@@ -0,0 +1,1251 @@
...
+}
+
+static int qcom_flash_setup_leds_device(struct qcom_flash_device *leds_dev,
+					struct qcom_flash_config *cfg,
+					struct device *dev)
+{
+	int rc;
+
+	leds_dev->dev = dev;
+
```
Comment (okias) [inline drivers/leds/flash/leds-qcom-spmi-flash.c:1062]: unneeded newline?

Comment (alikates) [reply drivers/leds/flash/leds-qcom-spmi-flash.c:1062]: The patches are from @nfraprado and he was gonna submit a new version to upstream, so maybe he wants to know about these issues.

File: Documentation/devicetree/bindings/leds/leds-qcom-spmi-flash.yaml (hunk 1 above)
Comment (alikates) [reply Documentation/devicetree/bindings/leds/leds-qcom-spmi-flash.yaml:1]: Same as below

**Comment by alikates** [comment]
Closing this because we can already use the LPG driver from Bjorn Andersson.

**Pull request closed without merging**